File: app/src/main/java/com/stho/software/nyota/utilities/LocationStabilityCounter.java
package com.stho.software.nyota.utilities;

public class LocationStabilityCounter {
    private static final int REQUIRED_COUNTER_VALUE = 10;
    private int counter = 0;

    public void reset() {
        counter = 0;
    }

    public boolean isStable() {
        if (counter > REQUIRED_COUNTER_VALUE) {
            return true;
        }
        else {
            counter++;
            return false;
        }
    }
}

File: app/src/main/java/com/stho/nyota/sky/utilities/LocationStabilityCounter.kt
package com.stho.nyota.sky.utilities

class LocationStabilityCounter {
    private var counter = 0
    fun reset() {
        counter = 0
    }

    val isStable: Boolean
        get() = if (counter > REQUIRED_COUNTER_VALUE) {
            true
        } else {
            counter++
            false
        }

    companion object {
        private const val REQUIRED_COUNTER_VALUE = 10
    }
}